lib/posix/nanosleep.c

```diff
@@ -48,7 +49,7 @@ int nanosleep(const struct timespec *rqtp, struct timespec *rmtp)
 	}
 
 	/* currently we have no mechanism to achieve greater resolution */
-	k_busy_wait(ns / NSEC_PER_USEC);
+	k_busy_wait(z_tmcvt(ns, Z_HZ_ns, Z_HZ_us, false, true, true, false));
```

Comment:
That doesn't look maintainable.  Why not:

    k_busy_wait(ceiling_fraction(ns, NSEC_PER_USEC);

`ceiling_fraction` is in `<sys/util.h>` if that's not already included.

Reply:
Sure.